File: GeneratedEntityReaderClasses/Components/DieIfSpeedBelowComponent.cs
namespace GeneratedNoitaClasses;
public class DieIfSpeedBelowComponent : NoitaComponentBase,  iNoitaType<DieIfSpeedBelowComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat min_speed { get; set; } = new();
	public NoitaFloat mMinSpeedSquared { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		min_speed.Read(s);
		mMinSpeedSquared.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		min_speed.Write(s);
		mMinSpeedSquared.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/DieIfSpeedBelowComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class DieIfSpeedBelowComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    min_speed:NoitaFloat = field(default_factory=NoitaFloat)
    mMinSpeedSquared:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.min_speed.Read(s)
        self.mMinSpeedSquared.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.min_speed.Write(s)
        self.mMinSpeedSquared.Write(s)

